Proofs/Experiments/MemoryAliasing.lean

```diff
@@ -214,8 +215,9 @@ theorem overlapping_read_test_1 {out : BitVec (16 * 8)}
     read_mem_bytes 16 src_addr s = out := by
   simp only [memory_rules] at h ⊢
   simp_mem
+  simp only [Nat.reduceMul, Nat.sub_self, BitVec.extractLsBytes_eq_self, BitVec.cast_eq]
```

Comment:
Just checking: if you have `(config := {useOmegaToClose := true})` in the `simp_mem` above, would you still need this line?

Reply:
Also, where does `Classical.choice` in `overlapping_read_test_1` come from?

Reply:
@shigoel 

- Yes, it'll be closed automatically given `(config := {useOmegaToClose := true})`
- @shigoel Lean's internal tactics very very freely use classical principles so I could imagine that `simp` internally winds up using `Classical.choice`. IMO, it doesn't really matter, as it's an axiom that's baked into Lean.